pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/DockerPipelineFromDockerfileScript.groovy

```diff
@@ -39,12 +40,26 @@ public class DockerPipelineFromDockerfileScript extends DeclarativeAgentScript<D
 
     @Override
     public Closure run(Closure body) {
-        String targetLabel = describable.label
-        if (targetLabel == null) {
-            targetLabel = script.dockerLabel()?.trim()
-        }
+        String targetLabel = script.declarativeProps(property: DeclarativePropsStep.Property.LABEL,
+            defaultValue: describable.label)
         LabelScript labelScript = (LabelScript) Label.DescriptorImpl.instanceForName("label", [label: targetLabel]).getScript(script)
         return labelScript.run {
+            String registryUrl = script.declarativeProps(property: DeclarativePropsStep.Property.REGISTRY_URL,
+                defaultValue: describable.registryUrl)
+            String registryCreds = script.declarativeProps(property: DeclarativePropsStep.Property.REGISTRY_CREDENTIALS,
+                defaultValue: describable.registryCredentialsId)
+            if (registryUrl != null) {
+                script.getProperty("docker").withRegistry(registryUrl, registryCreds) {
```

Comment:
How so? As I mentioned, I'm still not sure how to test registries at all, really.